### PR TITLE
WebDriver: Retry fetching content description

### LINKED ISSUE #1936: WebDriver: IllegalStateException: Could not retrieve hierarchy through maestro.getContentDescription()
### Is there an existing issue for this?

- [X] I have searched the existing issues and didn't find mine.

### Steps to reproduce

In our case the crash happens after pressing a login button after entering username/password. I'll debug the issue and either try to provide a patch or a public reproducible test case.

### Actual results

We started to see a crash in `WebDriver`, but so far we haven't found the root cause yet.

```
java.lang.IllegalStateException: Could not retrieve hierarchy through maestro.getContentDescription()
```

### Expected results

No crash

### About app

(not public website)

### About environment

* macOS 14.5
* Chrome Version 128.0.6613.85 (Official Build) (x86_64)

### Logs

<details>
<summary>Logs</summary>

```
java.lang.IllegalStateException: Could not retrieve hierarchy through maestro.getContentDescription()
        at maestro.drivers.WebDriver.contentDescriptor(WebDriver.kt:175)
        at maestro.ViewHierarchy$Companion.from-8JJjmZI(ViewHierarchy.kt:29)
        at maestro.utils.ScreenshotUtils$Companion.viewHierarchy-c1iYVAs(ScreenshotUtils.kt:99)
        at maestro.utils.ScreenshotUtils$Companion.waitForAppToSettle-OpTFbEc(ScreenshotUtils.kt:58)
        at maestro.utils.ScreenshotUtils$Companion.waitForAppToSettle-OpTFbEc$default(ScreenshotUtils.kt:38)
        at maestro.drivers.WebDriver.waitForAppToSettle-5RpGHs4(WebDriver.kt:366)
        at maestro.Maestro.waitForAppToSettle-5RpGHs4(Maestro.kt:474)
        at maestro.Maestro.waitForAppToSettle-5RpGHs4$default(Maestro.kt:469)
        at maestro.Maestro.hierarchyBasedTap-hbl3e4M(Maestro.kt:353)
        at maestro.Maestro.performTap-hbl3e4M(Maestro.kt:281)
        at maestro.Maestro.tap-BUbHBYE(Maestro.kt:195)
        at maestro.orchestra.Orchestra.tapOnElement(Orchestra.kt:777)
        at maestro.orchestra.Orchestra.executeCommand(Orchestra.kt:240)
        at maestro.orchestra.Orchestra.executeSubflowCommands(Orchestra.kt:605)
        at maestro.orchestra.Orchestra.runSubFlow(Orchestra.kt:646)
        at maestro.orchestra.Orchestra.runFlowCommand(Orchestra.kt:514)
        at maestro.orchestra.Orchestra.executeCommand(Orchestra.kt:270)
        at maestro.orchestra.Orchestra.executeSubflowCommands(Orchestra.kt:605)
        at maestro.orchestra.Orchestra.runSubFlow(Orchestra.kt:646)
        at maestro.orchestra.Orchestra.runFlowCommand(Orchestra.kt:514)
        at maestro.orchestra.Orchestra.executeCommand(Orchestra.kt:270)
        at maestro.orchestra.Orchestra.executeSubflowCommands(Orchestra.kt:605)
        at maestro.orchestra.Orchestra.runSubFlow(Orchestra.kt:646)
        at maestro.orchestra.Orchestra.runFlowCommand(Orchestra.kt:514)
        at maestro.orchestra.Orchestra.executeCommand(Orchestra.kt:270)
        at maestro.orchestra.Orchestra.executeSubflowCommands(Orchestra.kt:605)
        at maestro.orchestra.Orchestra.runSubFlow(Orchestra.kt:646)
        at maestro.orchestra.Orchestra.runFlowCommand(Orchestra.kt:514)
        at maestro.orchestra.Orchestra.executeCommand(Orchestra.kt:270)
        at maestro.orchestra.Orchestra.executeCommands(Orchestra.kt:201)
        at maestro.orchestra.Orchestra.runFlow(Orchestra.kt:111)
        at maestro.orchestra.Orchestra.runFlow$default(Orchestra.kt:75)
        at maestro.cli.runner.MaestroCommandRunner.runCommands(MaestroCommandRunner.kt:185)
        at maestro.cli.runner.TestRunner$runSingle$result$1.invoke(TestRunner.kt:54)
        at maestro.cli.runner.TestRunner$runSingle$result$1.invoke(TestRunner.kt:51)
        at maestro.cli.runner.TestRunner.runCatching(TestRunner.kt:145)
        at maestro.cli.runner.TestRunner.runSingle(TestRunner.kt:51)
        at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1$1.invoke(TestCommand.kt:291)
        at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1$1.invoke(TestCommand.kt:249)
        at maestro.cli.session.MaestroSessionManager.newSession(MaestroSessionManager.kt:107)
        at maestro.cli.session.MaestroSessionManager.newSession$default(MaestroSessionManager.kt:58)
        at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1.invokeSuspend(TestCommand.kt:249)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
        at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)```

</details>


### Maestro version

1.37.9

### How did you install Maestro?

install script (https://get.maestro.mobile.dev)

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pocmo**
## Proposed changes

This is a workaround for the `WebDriver` problem described in #1936. We are filling a login form and press the login button. During that navigation there's a short time where the content description is `null` which results in a forced crash.

With this workaround we try again to fetch the content description for a limited time until we finally give up.

## Testing

Tested locally with Chrome `128.0.6613.85`.

## Issues fixed

Closes #1936.

### CODE REVIEW

File: maestro-client/src/main/java/maestro/drivers/WebDriver.kt [hunk 1]
```diff
@@ -171,8 +171,19 @@ class WebDriver(val isStudio: Boolean) : Driver {
...
+        // there are edge cases where executeJS returns null, and we cannot get the hierarchy. In this situation
+        // we retry multiple times until throwing an error eventually.
+        var contentDesc: Any? = null
+        var retry = 0
+        while (contentDesc == null) {
+            contentDesc = executeJS("return window.maestro.getContentDescription()")
+            if (contentDesc == null) {
+                retry++
+            }
+            if (retry == 10) {
```
Comment (bartekpacia) [inline maestro-client/src/main/java/maestro/drivers/WebDriver.kt:183]: 1. Let's extract `10` to some constant.
2. Let's add issue link to the comment for context. E.g. `See issue #1936`

**Review by bartekpacia — APPROVED** [review]
This LGTM! Thanks

**Comment by pocmo** [comment]
Thank you. I updated the PR.

**Comment by bartekpacia** [comment]
Thanks, LGTM.

**Pull request merged**